src/main/kotlin/no/nav/syfo/aktivitetskrav/kafka/ExpiredVarselProducer.kt

```diff
@@ -0,0 +1,28 @@
+package no.nav.syfo.aktivitetskrav.kafka
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import org.apache.kafka.clients.producer.KafkaProducer
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.slf4j.LoggerFactory
+import java.util.*
+
+class ExpiredVarselProducer(private val producer: KafkaProducer<String, ExpiredVarsel>) {
+
+    suspend fun publishExpiredVarsel(expiredVarsel: ExpiredVarsel) {
+        try {
+            val record = ProducerRecord(TOPIC, UUID.randomUUID().toString(), expiredVarsel)
```

Comment:
Vi bør kanskje tenke gjennom om vi skal bruke noe annet som key her eller om random UUID er greit.

Reply:
Ja, hvis det er viktig at recordene konsumeres i samme rekkefølge som de produseres må man kanskje gjøre noe med dette. Men det blir vel ikke flere samtidige records for samme personIdent her? Så kanskje ikke nødvendig å gjøre noe med det i denne omgang.

Reply:
Ser nå at det er forskjellig fra producer til producer 🤔 Hva brukes dette typisk til?
Og hvordan blir for eksempel denne seende ut? 🤔
`val key = UUID.nameUUIDFromBytes(kafkaAktivitetskravVurdering.personIdent.toByteArray()).toString()`

Reply:
Så vidt jeg har skjønt så vil records med samme key havne på samme partisjon i den rekkefølgen de ble produsert - https://medium.com/event-driven-utopia/understanding-kafka-topic-partitions-ae40f80552e8
Men som Geir sier så er nok ikke det så viktig i dette tilfellet.

Reply:
Keyen brukes også ved [log compaction](https://docs.confluent.io/kafka/design/log_compaction.html). Men vi har vel evig retention på topicene våre, så da har det ikke så mye å si.